source/models/prediction.py

```diff
@@ -43,18 +42,24 @@ def value_check(feature_dict):
             print("____________________________________________")
 
             try:
-                if float(user_input) >= value['min'] and float(user_input) <= value['max']:
+                if (
+                    float(user_input) >= value["min"]
+                    and float(user_input) <= value["max"]
+                ):
                     edited_input.append(user_input)
                 else:
                     raise ValueError()
 
             except ValueError:
-                print(f"Please enter a Float that's between {value['min']} and {value['max']}")
+                print(
+                    f"Please enter a Float that's between {value['min']} and {value['max']}"
+                )
                 break
 
     return edited_input
 
-def prediction(user_input, model_encoder_dictionary):
+
+def prediction(user_input: list, model_encoder_dictionary: dict) -> None:
```

Comment:
A better name for `user_input` could be `features`.  
Also, in the docstring you mention that it should be a 4-element list, but it's not really true.

Reply:
changed to features_input to signify that it's the features that you inputted